09_json/n0npax/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	puppy "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy/store"
+)
+
+var (
+	logFatalf = log.Fatal
+	parser    = parseArgs
+)
+
+func main() {
+	config, err := parser(os.Args[1:])
+	if err != nil {
+		logFatalf(err)
+	}
+	logFatalf(runPuppyServer(&config))
+}
+
+func runPuppyServer(c *config) error {
```

Comment:
This isn't really a server (as in something continuously running and serving), right?
Maybe `processPuppies` or similar?

I think I'd find my way around it more easily if it read something akin to:

```go
store, err := createStorer(cfg)
if err != nil {
    return err
}
if err := feedStorer(store, cfg.puppyReader); err != nil {
     return err
}
return nil